ssl/record/ssl3_record.c

```diff
@@ -1624,6 +1625,7 @@ int dtls1_process_record(SSL *s, DTLS1_BITMAP *bitmap)
      *    1: Success or MTE decryption failed (MAC will be randomised)
      */
     if (enc_err == 0) {
+        ERR_pop_to_mark();
```

Comment:
I'm failing to grasp the logic behind this. Why do we want to clear the errors in the event that the record is publicly invalid? Surely the errors should be interesting in that case?

Reply:
Ah! I've now read the commit message - so this now makes more sense. It might be helpful to have some of that text as a comment here?

Reply:
comment added